lib/Data/Validate/Sanctions.pm

```diff
@@ -119,28 +121,52 @@ sub get_sanctioned_info {    ## no critic (RequireArgUnpacking)
 
             my $checked_dob;
 
-            $matched_name = $sanctioned_name;
-            $matched_file = $file;
-
             # Some clients in sanction list can have more than one date of birth
-            # Comparison is made using the epoch value
-            my $client_dob_epoch   = Date::Utility->new($date_of_birth)->epoch;
-            my $sanctions_dob_list = $data->{$file}->{names_list}->{$sanctioned_name}->{dob_epoch};
-
-            # If the dob_epoch is missing from the sanctions.yml, automatically mark
-            # the client as a terrorist, regardless of further checks
-            unless (@$sanctions_dob_list) {
-                $dob_missing = 1;
+            # Comparison is made using the epoch and year values
+            my $client_dob_date  = Date::Utility->new($date_of_birth);
+            my $client_dob_epoch = $client_dob_date->epoch;
+            my $client_dob_year  = $client_dob_date->year;
+
+            my $sanctions_epoch_list = $data->{$file}->{names_list}->{$sanctioned_name}->{dob_epoch} // [];
+            my $sanctions_year_list  = $data->{$file}->{names_list}->{$sanctioned_name}->{dob_year}  // [];
+
+            # Saving names with dob_text for later check.
+            my $has_no_epoch_or_year = @$sanctions_epoch_list || @$sanctions_year_list ? 0 : 1;
+            my $has_dob_text = @{$data->{$file}->{names_list}->{$sanctioned_name}->{dob_text} // []} ? 1 : 0;
+            if ($has_dob_text || $has_no_epoch_or_year) {
+                push @match_with_dob_text,
+                    {
+                    name => $sanctioned_name,
+                    file => $file
+                    };
             }
 
-            $checked_dob = any { $_ eq $client_dob_epoch } @{$sanctions_dob_list};
+            $checked_dob = any { $_ eq $client_dob_epoch } @{$sanctions_epoch_list};
+            return _possible_match($file, $sanctioned_name, 'Date of birth matches', $date_of_birth) if $checked_dob;
 
-            return _possible_match($matched_file, $matched_name, 'Date of birth matches', $date_of_birth) if $checked_dob;
+            $checked_dob = any { $_ eq $client_dob_year } @{$sanctions_year_list} unless $checked_dob;
+            return _possible_match($file, $sanctioned_name, 'Year of birth matches', $client_dob_year) if $checked_dob;
         }
     }
 
     # Return a possible match if the name matches and no date of birth is present in sanctions
-    return _possible_match($matched_file, $matched_name, 'Name is similar', 'N/A') if ($matched_name && $dob_missing);
+    for my $match (@match_with_dob_text) {
+        # We match only in case we have full match for the name
+        # in other case we may get to many false positive
+        my ($sacntion_name, $client_name) = map { uc(s/[^[:alpha:]\s]//gr) } ($match->{name}, $client_full_name);
```

Comment:
There's no need to match names here, because client and sanction entry names have been already matched in [L120](https://github.com/binary-com/perl-Data-Validate-Sanctions/pull/33/files#diff-7e74c4e3dc4bc7a8c2b436610581ade9R120), haven't they?

Reply:
as discussed in slack, we need here to check  for 100% match.